model/Check/System/FileSystemS3CachePathCheck.php

```diff
@@ -0,0 +1,139 @@
+ *
+ * @package oat\taoSystemStatus\model\Check\System
+ * @author Aleh Hutnikau, <aleh@taotesting.com>
+ */
+class FileSystemS3CachePathCheck extends AbstractCheck
+{
+
+    /**
+     * @param array $params
+     * @return Report
+     */
+    public function __invoke($params = []): Report
+    {
+        if (!$this->isActive()) {
+            return new Report(Report::TYPE_INFO, 'Check ' . $this->getId() . ' is not active');
+        }
+        $report = $this->checkFilesystemCaches();
+        return $this->prepareReport($report);
+    }
+
+    /**
+     * @return bool
+     */
+    public function isActive(): bool
+    {
+        $fileSystem = $this->getFileSystemService();
+        return class_exists('oat\awsTools\AwsFileSystemService')
+            && get_class($fileSystem) === 'oat\awsTools\AwsFileSystemService';
+    }
+
+    /**
+     * @return string
+     */
+    public function getType(): string
+    {
+        return self::TYPE_SYSTEM;
+    }
+
+    /**
+     * @return string
+     */
+    public function getCategory(): string
+    {
+        return __('TAO Configuration');
+    }
+
+    /**
+     * @return string
+     */
+    public function getDetails(): string
+    {
+        return __('Cache directory path on S3 file system adapters');
+    }
+
+    /**
+     * @return Report
+     */
+    private function checkFilesystemCaches() : Report
+    {
+        $adaptersWithCacheInsideTaoRoot = [];
+        $taoRootDir = $this->getTaoRootDir();
+        $adapters = $this->getFileSystemService()->getOption(FileSystemService::OPTION_ADAPTERS);
+        foreach ($adapters as $adapterId => $adapterConfig) {
+            if (!isset($adapterConfig['options'])) {
+                continue;
+            }
+            $options = array_pop($adapterConfig['options']);
+            if (!isset($options['cache'])) {
+                continue;
+            }
+            $cachePath = realpath($options['cache']);
+            if (strpos($cachePath, $taoRootDir) === 0) {
+                $adaptersWithCacheInsideTaoRoot[] = $adapterId;
+            }
+        }
+
+        if (!empty($adaptersWithCacheInsideTaoRoot)) {
+            return new Report(
+                Report::TYPE_WARNING,
+                __('Cache directory is inside tao root directory. Filesystem adapters: %s', implode(', ', $adaptersWithCacheInsideTaoRoot))
+            );
+        }
+
+        return new Report(Report::TYPE_SUCCESS, __('Cache directories of filesystem adapters correctly configured.'));
+    }
+
+    /**
+     * @return FileSystemService
+     */
+    private function getFileSystemService() : FileSystemService
+    {
+        /** @noinspection PhpIncompatibleReturnTypeInspection */
+        return $this->getServiceLocator()->get(FileSystemService::SERVICE_ID);
+    }
+
+    /**
+     * @return string
+     */
+    private function getTaoRootDir(): string
+    {
+        return realpath(rtrim(ROOT_PATH, '\\/').DIRECTORY_SEPARATOR.'..'.DIRECTORY_SEPARATOR);
```

Comment:
I do not see the value of that. 

In that case we will have to calculate root path twice:
1. in the migration (to register check with that parameter to be injected to __invoke)
2. in the `\oat\taoSystemStatus\scripts\install\RegisterChecks` file to register that check during install.

So that means that we will need to repeat this line twice:
```
realpath(rtrim(ROOT_PATH, '\\/').DIRECTORY_SEPARATOR.'..'.DIRECTORY_SEPARATOR);
```

Or we can move it to static method of `FileSystemS3CachePathCheck`. But it that case what is the value of that change if that method still be in `FileSystemS3CachePathCheck`?

Reply:
When you have unit tests the value becomes clear: no more global dependencies once it's properly injected.
Indeed, as here there are no unit tests it doesn't add much, for the time being. 
My thoughts on the duplication: obviously the "path logic" shouldn't be duplicated but rather extracted in a config and have that config referenced where it's required.